File: queues5.c
#include <stdio.h>
#include <string.h>

struct Request {
    int id;
    int priority;
};

struct Queue {
    struct Request arr[1000];
    int front, rear, size;
};

void initQueue(struct Queue *q) {
    q->front = -1;
    q->rear = -1;
    q->size = 1000;
}

int isFull(struct Queue *q) {
    return ((q->rear + 1) % q->size) == q->front;
}

int isEmpty(struct Queue *q) {
    return q->front == -1;
}

void enqueue(struct Queue *q, int id, int priority) {
    if (isFull(q)) {
        printf("Queue is full\n");
        return;
    }

    if (isEmpty(q)) {
        q->front = q->rear = 0;
    } else {
        q->rear = (q->rear + 1) % q->size;
    }

    q->arr[q->rear].id = id;
    q->arr[q->rear].priority = priority;

    printf("Enqueued request ID: %d with priority: %d\n", id, priority);
}

void dequeue(struct Queue *q) {
    if (isEmpty(q)) {
        printf("Queue is empty\n");
        return;
    }

    printf("Processed request ID: %d with priority: %d\n",
           q->arr[q->front].id, q->arr[q->front].priority);

    if (q->front == q->rear) {
        q->front = q->rear = -1;
    } else {
        q->front = (q->front + 1) % q->size;
    }
}

void peek(struct Queue *q) {
    if (isEmpty(q)) {
        printf("Queue is empty\n");
    } else {
        printf("Front request ID: %d with priority: %d\n",
               q->arr[q->front].id, q->arr[q->front].priority);
    }
}

int main() {
    int n;
    scanf("%d", &n);

    struct Queue q;
    initQueue(&q);

    while (n--) {
        char command[20];
        scanf("%s", command);

        if (strcmp(command, "enqueue") == 0) {
            int id, priority;
            scanf("%d %d", &id, &priority);
            enqueue(&q, id, priority);
        }
        else if (strcmp(command, "dequeue") == 0) {
            dequeue(&q);
        }
        else if (strcmp(command, "peek") == 0) {
            peek(&q);
        }
    }

    return 0;
}

File: queues5.java
import java.util.Scanner;

class CircularQueue {
    private int[] arr;
    private int front;
    private int rear;
    private int count;
    private int capacity;

    public CircularQueue(int capacity) {
        this.capacity = capacity;
        arr = new int[capacity];
        front = 0;
        rear = -1;
        count = 0;
    }

    public boolean isFull() {
        return count == capacity;
    }

    public boolean isEmpty() {
        return count == 0;
    }

    public void enqueue(int value) {
        if (isFull()) {
            System.out.println("Queue is full");
            return;
        }
        rear = (rear + 1) % capacity;
        arr[rear] = value;
        count++;
        System.out.println("Enqueued request ID: " + value);
    }

    public void dequeue() {
        if (isEmpty()) {
            System.out.println("Queue is empty");
            return;
        }
        System.out.println("Processed request ID: " + arr[front]);
        front = (front + 1) % capacity;
        count--;
    }

    public void display() {
        if (isEmpty()) {
            System.out.println("Queue is empty");
            return;
        }
        System.out.print("Queue elements: ");
        for (int i = 0; i < count; i++) {
            int index = (front + i) % capacity;
            System.out.print(arr[index] + " ");
        }
        System.out.println();
    }

    public void size() {
        System.out.println("Current queue size: " + count);
    }
}

public class TaskSchedulerQueue {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();         // number of operations
        int capacity = sc.nextInt();  // queue capacity
        sc.nextLine(); // consume remaining newline

        CircularQueue queue = new CircularQueue(capacity);

        for (int i = 0; i < n; i++) {
            String line = sc.nextLine().trim();
            String[] parts = line.split(" ");

            switch (parts[0]) {
                case "enqueue":
                    int value = Integer.parseInt(parts[1]);
                    queue.enqueue(value);
                    break;
                case "dequeue":
                    queue.dequeue();
                    break;
                case "display":
                    queue.display();
                    break;
                case "size":
                    queue.size();
                    break;
            }
        }

        sc.close();
    }
}
